record Dolby Vision at 60 fps, 1080p and 4K

Smali changes of vendor_oplus_camera 3238e53, over baksmali --sequential-labels output.

diff --git a/classes7/mm/DV.smali b/classes7/mm/DV.smali
index 11ad85297aded46d2957ab512706d40fae34a95c..c72def07bbc0784770e6e64d0b0e5428329e6210 100644
--- a/classes7/mm/DV.smali
+++ b/classes7/mm/DV.smali
@@ -469,10 +469,10 @@
 .method public static in(Lmm/i;)V
     .locals 3
 
-    .line 24
+    .line 26
     iget-object v0, p0, Lmm/i;->h0:Landroid/media/MediaFormat;
 
-    .line 25
+    .line 27
     if-eqz v0, :cond_28
 
     invoke-static {p0}, Lmm/DV;->dv(Lmm/i;)Z
@@ -483,7 +483,7 @@
 
     goto :goto_28
 
-    .line 26
+    .line 28
     :cond_b
     const-string p0, "profile"
 
@@ -491,34 +491,34 @@
 
     invoke-virtual {v0, p0, v1}, Landroid/media/MediaFormat;->setInteger(Ljava/lang/String;I)V
 
-    .line 27
+    .line 29
     const-string p0, "level"
 
     invoke-virtual {v0, p0}, Landroid/media/MediaFormat;->removeKey(Ljava/lang/String;)V
 
-    .line 28
+    .line 30
     const-string p0, "color-standard"
 
     const/4 v2, 0x6
 
     invoke-virtual {v0, p0, v2}, Landroid/media/MediaFormat;->setInteger(Ljava/lang/String;I)V
 
-    .line 29
+    .line 31
     const-string p0, "color-transfer"
 
     const/4 v2, 0x7
 
     invoke-virtual {v0, p0, v2}, Landroid/media/MediaFormat;->setInteger(Ljava/lang/String;I)V
 
-    .line 30
+    .line 32
     const-string p0, "color-range"
 
     invoke-virtual {v0, p0, v1}, Landroid/media/MediaFormat;->setInteger(Ljava/lang/String;I)V
 
-    .line 31
+    .line 33
     return-void
 
-    .line 25
+    .line 27
     :cond_28
     :goto_28
     return-void
@@ -527,7 +527,7 @@
 .method public static out(Lmm/i;Landroid/media/MediaFormat;)V
     .locals 3
 
-    .line 34
+    .line 36
     invoke-static {p0}, Lmm/DV;->dv(Lmm/i;)Z
 
     move-result v0
@@ -550,30 +550,30 @@
 
     goto :goto_29
 
-    .line 35
+    .line 37
     :cond_15
     const-string v0, "video/dolby-vision"
 
     invoke-virtual {p1, v1, v0}, Landroid/media/MediaFormat;->setString(Ljava/lang/String;Ljava/lang/String;)V
 
-    .line 36
+    .line 38
     const-string v0, "profile"
 
     const/16 v1, 0x100
 
     invoke-virtual {p1, v0, v1}, Landroid/media/MediaFormat;->setInteger(Ljava/lang/String;I)V
 
-    .line 37
+    .line 39
     const-string v0, "level"
 
     iget p0, p0, Lmm/i;->g1:I
 
     invoke-virtual {p1, v0, p0}, Landroid/media/MediaFormat;->setInteger(Ljava/lang/String;I)V
 
-    .line 38
+    .line 40
     return-void
 
-    .line 34
+    .line 36
     :cond_29
     :goto_29
     return-void
@@ -582,7 +582,7 @@
 .method public static sample(Lmm/i;ILjava/nio/ByteBuffer;Landroid/media/MediaCodec$BufferInfo;)Ljava/nio/ByteBuffer;
     .locals 6
 
-    .line 41
+    .line 43
     invoke-static {p0}, Lmm/DV;->dv(Lmm/i;)Z
 
     move-result v0
@@ -605,7 +605,7 @@
 
     goto :goto_4c
 
-    .line 42
+    .line 44
     :cond_15
     iget p0, p3, Landroid/media/MediaCodec$BufferInfo;->size:I
 
@@ -619,12 +619,12 @@
 
     move-result-object p0
 
-    .line 43
+    .line 45
     invoke-virtual {p2}, Ljava/nio/ByteBuffer;->duplicate()Ljava/nio/ByteBuffer;
 
     move-result-object p1
 
-    .line 44
+    .line 46
     iget p2, p3, Landroid/media/MediaCodec$BufferInfo;->offset:I
 
     iget v0, p3, Landroid/media/MediaCodec$BufferInfo;->size:I
@@ -639,7 +639,7 @@
 
     invoke-virtual {p2, v0}, Ljava/nio/ByteBuffer;->position(I)Ljava/nio/ByteBuffer;
 
-    .line 45
+    .line 47
     invoke-virtual {p0, p1}, Ljava/nio/ByteBuffer;->put(Ljava/nio/ByteBuffer;)Ljava/nio/ByteBuffer;
 
     move-result-object p1
@@ -652,7 +652,7 @@
 
     invoke-virtual {p1}, Ljava/nio/ByteBuffer;->flip()Ljava/nio/ByteBuffer;
 
-    .line 46
+    .line 48
     invoke-virtual {p0}, Ljava/nio/ByteBuffer;->limit()I
 
     move-result v2
@@ -667,24 +667,26 @@
 
     invoke-virtual/range {v0 .. v5}, Landroid/media/MediaCodec$BufferInfo;->set(IIJI)V
 
-    .line 47
+    .line 49
     return-object p0
 
-    .line 41
+    .line 43
     :cond_4c
     :goto_4c
     return-object p2
 .end method
 
 .method public static want(Lom/t0;)Z
-    .locals 2
+    .locals 5
 
     .line 19
     invoke-virtual {p0}, Lom/t0;->Se()Z
 
     move-result v0
 
-    if-eqz v0, :cond_24
+    const/4 v1, 0x0
+
+    if-eqz v0, :cond_38
 
     const-string v0, "com.oplus.feature.video.dv.support"
 
@@ -692,37 +694,64 @@
 
     move-result v0
 
-    if-eqz v0, :cond_24
+    if-nez v0, :cond_10
+
+    goto :goto_38
 
     .line 20
-    invoke-virtual {p0}, Lom/t0;->Qa()I
+    :cond_10
+    invoke-virtual {p0}, Lom/t0;->h7()Ljava/lang/String;
 
-    move-result v0
+    move-result-object v0
 
-    const/16 v1, 0x1e
+    .line 21
+    const-string v2, "video_size_1080p"
 
-    if-ne v0, v1, :cond_24
+    invoke-virtual {v2, v0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
 
-    const-string v0, "video_size_1080p"
+    move-result v2
 
-    invoke-virtual {p0}, Lom/t0;->h7()Ljava/lang/String;
+    .line 22
+    invoke-virtual {p0}, Lom/t0;->Qa()I
 
-    move-result-object p0
+    move-result v3
+
+    const/16 v4, 0x1e
+
+    if-ne v3, v4, :cond_24
 
-    invoke-virtual {v0, p0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+    move v1, v2
+
+    goto :goto_37
+
+    :cond_24
+    invoke-virtual {p0}, Lom/t0;->Qa()I
 
     move-result p0
 
-    if-eqz p0, :cond_24
+    const/16 v3, 0x3c
 
-    const/4 p0, 0x1
+    if-ne p0, v3, :cond_37
 
-    goto :goto_25
+    if-nez v2, :cond_36
 
-    :cond_24
-    const/4 p0, 0x0
+    const-string p0, "video_size_4kuhd"
+
+    invoke-virtual {p0, v0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+
+    move-result p0
+
+    if-eqz p0, :cond_37
+
+    :cond_36
+    const/4 v1, 0x1
+
+    :cond_37
+    :goto_37
+    return v1
 
     .line 19
-    :goto_25
-    return p0
+    :cond_38
+    :goto_38
+    return v1
 .end method
